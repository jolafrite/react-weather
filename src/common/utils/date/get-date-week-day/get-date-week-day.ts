const weekDays = [
  "sunday",
  "monday",
  "tuesday",
  "wednesday",
  "thursday",
  "friday",
  "saturday"
];

export const getDateWeekDay = (date: Date | string): string => {
  return weekDays[new Date(date).getDay()];
};
