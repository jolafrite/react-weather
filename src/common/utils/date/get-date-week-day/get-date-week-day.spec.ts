import { getDateWeekDay } from "./get-date-week-day";

describe('utils: getDateWeekDay', () => {

  describe('Object Date', () => {
    describe('2019-06-18', () => {
      test('should return tuesday', () => {
        expect(
          getDateWeekDay(new Date('2019-06-18'))
        ).toMatchSnapshot();
      });
    });

    describe('2019-06-16', () => {
      test('should return sunday', () => {
        expect(
          getDateWeekDay(new Date('2019-06-16'))
        ).toMatchSnapshot();
      });
    });
  });

  describe('String Date', () => {
    describe('2019-06-21', () => {
      test('should return friday', () => {
        expect(
          getDateWeekDay('2019-06-21')
        ).toMatchSnapshot();
      });
    });
  });

});
