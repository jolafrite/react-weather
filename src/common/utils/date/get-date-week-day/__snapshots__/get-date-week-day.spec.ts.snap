// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils: getDateWeekDay Object Date 2019-06-16 should return sunday 1`] = `"sunday"`;

exports[`utils: getDateWeekDay Object Date 2019-06-18 should return tuesday 1`] = `"tuesday"`;

exports[`utils: getDateWeekDay String Date 2019-06-21 should return friday 1`] = `"friday"`;
